Testing jest

sergiocortessat opened the pull request:
Pull request 1 To-Do-List jest test

Here I present the To-Do-List project on javascript tested with jest. 

 - The test cases involve only the methods that have a direct impact on the information. 
 - As such I have testes the functions in charge of selecting and altering dom elements, for this I created a file called domfortes.js in which we test the exact functionality of it using the jsdom npm library.
 - I have tested as well the primary constructor in charge of returning the main information that will be used to set local storage. As such and following good testing practices, I have tested such a module as it will guarantee me the proper return of the values needed to push the object to the dom and its respective local storage. As such the local storage always updates to its fullest within this application. Therefore testing the constructor ensures a more effective test

sergiocortessat committed:
Add jest installation

sergiocortessat committed:
Add test samples

sergiocortessat committed:
Add test case simple.test

sergiocortessat committed:
Add test to todoitem

sergiocortessat committed:
Add test to todolist

sergiocortessat committed:
Add test todo

sergiocortessat committed:
Add test file create todo

sergiocortessat committed:
Add domfortest .js

sergiocortessat committed:
Add domrfortest.test.js file

sergiocortessat committed:
Add babel.rc config files

sergiocortessat committed:
Add eliminate unnecesary files

sergiocortessat committed:
Add readme test part

sergiocortessat committed:
Add testing to constructor

mimipeshy commented:
## STATUS: CHANGES REQUESTED 🔴 ♻️ 
Hello, @sergiocortessat 👋🏾 👋🏾 congratulations on making it this far 😄 . 

#### Highlights

- Good job on testing this project. Your test cases pass ✔️ 
- Good job on the PR  ✔️ 

#### To improve

- [x] I ran into the following error when I tried to run your tests at first. Kindly add the missing module to your package.json file

![image](https://user-images.githubusercontent.com/33067256/121188941-42f34c00-c872-11eb-9cf5-937d1823cd72.png)

#### OPTIONAL 🟢 

- [x] To make your code neat, create a folder named `__tests__` and move all the tests into the folder.


That's all 😄, kindly submit another review when done
If you need any clarification, kindly post your comment here and tag me @mimipeshy and will be glad to help

Regards,
Peris Ndanu

sergiocortessat committed:
Add tse changes babel in jason.lock, __test__ folder

sergiocortessat committed:
Add new bundle

sergiocortessat committed:
Add folders organization

sergiocortessat commented:
Hello TSE.

Thank you very much for checking on my to-do list test. I have added the requested changes such as:

- Added the @babel/preset-env to the lock.json file
- Added correspondent folders that better divide the assets, javascript, and __test__

arslanbisharat reviewed (changes requested):
#Approved 👏 

Hi @sergiocortessat , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

sergiocortessat committed:
Add files

sergiocortessat committed:
Add test corrections

sergiocortessat committed:
Add test createToDo
